### PR TITLE
Add ssh with field mapping and example log

### PR DESCRIPTION
**Pull request by JW-Corelight**

### CODE REVIEW

**Review by irakledibm — DISMISSED** [review]

**Comment by mikeradka** [comment]
Could you add in the Markdown to which OCSF version this is mapped, and have this example listed under a subfolder for that version? There are a few examples of this in the mappings directory, and I am going to update our Windows rc2 mappings in accordance with this

**Comment by JW-Corelight** [comment]
@mikeradka The first static mapping is: 

> | OCSF field                          | Value                                           |
| ----------------------------------- | ----------------------------------------------- |
| `metadata.version`                  | "1.3.0"     |

**Comment by mikeradka** [comment]
> @mikeradka Maybe you're saying change the directory structure to include the OCSF version, so for this folder, the directory would be:
> 
> `ocsf/examples/tree/main/mappings/markdown/Zeek/ssh/v1.3.0/`
> 
> Right?

That's right. Also for the README, was thinking something like this to be consistent with other markdown:

image

[here is a link to that example](https://github.com/ocsf/examples/blob/main/mappings/markdown/Microsoft/Windows%20Events/4624/README.md)

**Comment by JW-Corelight** [comment]
@mikeradka I think I've done this - can you verify?

**Comment by mikeradka** [comment]
> @mikeradka I think I've done this - can you verify?

I think this looks good. If compared to the AWS folders, there is a slight difference - the `version` subfolder comes before the `log_name` subfolder whereas i think yours has that reversed. Either way works for me as long as we do our best to keep the structure uniform.

**Review by mikeradka — APPROVED** [review]

**Comment by mikeradka** [comment]
@JW-Corelight the field mappings are looking pretty good. The Zeek descriptions have been very helpful for me. I did notice when validating the translated sample that there are some validation errors. Looks like a few required fields missing and datatype mismatches:

```
 "errors": [
    {
      "error": "attribute_wrong_type",
      "message": "Attribute \"client_hassh.fingerprint\" value has wrong type; expected fingerprint (object), got string_t.",
      "value": "ec7378c1a92f5a8dde7e8b7a1ddf33d1",
      "attribute": "fingerprint",
      "value_type": "string_t",
      "attribute_path": "client_hassh.fingerprint",
      "expected_type": "fingerprint (object)"
    },
    {
      "error": "attribute_required_missing",
      "message": "Required attribute \"connection_info.direction_id\" is missing.",
      "attribute": "direction_id",
      "attribute_path": "connection_info.direction_id"
    },
    {
      "error": "attribute_wrong_type",
      "message": "Attribute \"metadata.logged_time\" value has wrong type; expected timestamp_t (long_t), got string_t.",
      "value": "2024-10-16T04:54:04.950981Z",
      "attribute": "logged_time",
      "value_type": "string_t",
      "attribute_path": "metadata.logged_time",
      "expected_type": "timestamp_t"
    },
    {
      "error": "attribute_required_missing",
      "message": "Required attribute \"metadata.loggers[0].product.vendor_name\" is missing.",
      "attribute": "vendor_name",
      "attribute_path": "metadata.loggers[0].product.vendor_name"
    },
    {
      "error": "attribute_wrong_type",
      "message": "Attribute \"start_time\" value has wrong type; expected timestamp_t (long_t), got string_t.",
      "value": "2024-10-16T04:54:04.474340Z",
      "attribute": "start_time",
      "value_type": "string_t",
      "attribute_path": "start_time",
      "expected_type": "timestamp_t"
    },
    {
      "error": "attribute_wrong_type",
      "message": "Attribute \"server_hassh.fingerprint\" value has wrong type; expected fingerprint (object), got string_t.",
      "value": "b12d2871a1189eff20364cf5333619ee",
      "attribute": "fingerprint",
      "value_type": "string_t",
      "attribute_path": "server_hassh.fingerprint",
      "expected_type": "fingerprint (object)"
    },
    {
      "error": "attribute_wrong_type",
      "message": "Attribute \"time\" value has wrong type; expected timestamp_t (long_t), got string_t.",
      "value": "2024-10-16T04:54:04.474340Z",
      "attribute": "time",
      "value_type": "string_t",
      "attribute_path": "time",
      "expected_type": "timestamp_t"
    },
    {
      "error": "attribute_wrong_type",
      "message": "Attribute \"protocol_ver\" value has wrong type; expected string_t, got integer_t (integer in range of -2^63 to 2^63 - 1).",
      "value": 2,
      "attribute": "protocol_ver",
      "value_type": "integer_t",
      "attribute_path": "protocol_ver",
      "expected_type": "string_t"
    },
    {
      "error": "attribute_required_missing",
      "message": "Required attribute \"type_uid\" is missing.",
      "attribute": "type_uid",
      "attribute_path": "type_uid"
    }
  ],
  "error_count": 9
```

**Review by irakledibm — CHANGES_REQUESTED** [review]
time, start_time and logged_time are of type timestamp_t, which is Long data type. An example is showing this format: "logged_time": "2024-10-16T04:18:12.066275Z", corresponding to datetime_t, which is incorrect. Value have to be converted to a numeric representation.

**Comment by JW-Corelight** [comment]
I do have all the original type information, so i can add that column for context. it seems like it's helpful

**Comment by JW-Corelight** [comment]
I did a pretty significant review after your feedback.  I think I've completed everything, plus reorganized and removed some unnecessary fields for clarity.

**Comment by JW-Corelight** [comment]
I've updated the sample log that is converted to OCSF given these updated instructions

**Comment by mikeradka** [comment]
> I've updated the sample log that is converted to OCSF given these updated instructions

@JW-Corelight this is looking great! Just one mismatched data type to clear up, `class_uid`, and I think this could be ready for the races.

```
{
  "uid": "CwjjT11TKp4IDDkiol",
  "warnings": [],
  "errors": [
    {
      "error": "attribute_wrong_type",
      "message": "Attribute \"class_uid\" value has wrong type; expected integer_t, got string_t.",
      "value": "4007",
      "attribute": "class_uid",
      "value_type": "string_t",
      "attribute_path": "class_uid",
      "expected_type": "integer_t"
    }
  ],
  "error_count": 1,
  "warning_count": 0
}
```

File: mappings/markdown/Zeek/v1.3.0/ssh_log/samples/ssh_log.ocsf [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    ],
+    "log_name": "ssh",
+    "uid": "CwjjT11TKp4IDDkiol",
+    "product": {
+      "name": "Zeek",
+      "vendor_name": "Zeek"
+    }
+  },
+  "category_uid": "4",
+  "class_uid": "4007",
```
Comment (mikeradka) [inline mappings/markdown/Zeek/v1.3.0/ssh_log/samples/ssh_log.ocsf:18]: This is the last remaining validation issue, the value should be changed to an integer.

**Review by mikeradka — CHANGES_REQUESTED** [review]

**Review by mikeradka — APPROVED** [review]
Looks great @JW-Corelight

**Review by irakledibm — APPROVED** [review]

File: mappings/markdown/Zeek/v1.3.0/ssh_log/README.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+| `client_hassh.fingerprint.algorithm_id` | `hassh`  | The hassh information.                                              | If `hassh` is present, then "1". (MD5) <br>Type is Integer.                             |
+| `server_hassh.fingerprint.algorithm_id` | `hasshServer` | The hasshServer information.                                   | If `hasshServer` is present, then "1". (MD5) <br>Type is Integer.                       |
+
+ ### Unmapped:
+| OCSF       | Raw               | Zeek Field Description                                              |
+| ---------- | ----------------- | ------------------------------------------------------------------- |
+| `unmapped` | `hasshVersion`    | The hasshVersion information.                                       |
+| `unmapped` | `hasshAlgorithms` | The hasshAlgorithms information.                                    |
+| `unmapped` | `hasshServerAlgorithms` | The hasshServerAlgorithms information.                        |
```
Comment (Aniak5) [inline mappings/markdown/Zeek/v1.3.0/ssh_log/README.md:70]: Is `haasshAlgorithms/hasshServerAlgorithms ` a list of algorithms? Or could we use `client_hassh.fingerprint.algorithm` / `server_hassh.fingerprint.algorithm`?

**Review by Aniak5 — APPROVED** [review]
LGTM!

File: mappings/markdown/Zeek/v1.3.0/ssh_log/README.md (hunk 2 above)
Comment (JW-Corelight) [reply mappings/markdown/Zeek/v1.3.0/ssh_log/README.md:70]: From the [OCSF HASSH](https://schema.ocsf.io/1.3.0/objects/hassh) page:

> algorithm | recommended | String | The concatenation of key exchange, encryption, authentication and compression algorithms (separated by ';').
> -- | -- | -- | --

**Pull request merged**